Add support for Skip-EWS GitHub label

aj062 opened the pull request:
#### 57ba798e7aa4dfde86461fa927fae24d40486fe3
<pre>
Add support for Skip-EWS GitHub label
<a href="https://bugs.webkit.org/show_bug.cgi?id=247900">https://bugs.webkit.org/show_bug.cgi?id=247900</a>

Reviewed by Jonathan Bedard.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin):
(GitHubMixin._does_pr_has_skip_label): Method to check if pr has skip-ews label.
(ValidateChange.__init__):
(ValidateChange):
(ValidateChange.start):
(ValidateChange.validate_github):

Canonical link: <a href="https://commits.webkit.org/256760@main">https://commits.webkit.org/256760@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6fad17f9c06b4a4b9bbbf0becdd89668a228fbc3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96778 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6043 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29870 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106305 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/166595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100757 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6255 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34774 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89163 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/103008 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102453 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/4678 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83391 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/31642 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/100369 "webkitpy-tests (failure)") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/86518 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88391 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/74582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/79 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19877 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/95731 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/71 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/21295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/4860 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43821 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2254 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1297 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/40583 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
It's unclear to me what "skip-ews" means. Does it mean do not start EWS runs at all because the PR is not ready? does it mean something else? Seems to me like this is something about merge-queue, which was not what I first thought it would be.

JonWBedard reviewed (commented):
On the one hand, this PR does literally do what it says (ie, add a label called `skip-news` that will prevent all EWS from running on a PR)

On the other hand, I'm still not convinced we have unity on how skipping EWS should work. This PR is one option, another option is to reconsider what we mean by a "draft" pull-request. I have thought of that as a PR that doesn't set `r?`, but we could consider having "draft" mean "no EWS" and then have a specific "ews" PR that triggers EWS. This could also have the benefit of functioning as a retry, albeit a global retry instead of a single queue retry. I've previously said I'm against not running EWS on draft PRs, but I think having a specific "ews" label would address my concerns.

JonWBedard reviewed (approved):
We discussed this on Slack, and agreed that we liked this design, although with the added feature of `git-webkit pr --draft` automatically adding the `skip-ews` label.

JonWBedard commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -158,6 +158,7 @@ class GitHubMixin(object):
     addURLs = False
     pr_open_states = ['open']
     pr_closed_states = ['closed']
+    SKIP_EWS_LABEL = 'skip-ews'
```
Don't we need to change our hook processing so that _removing_ this label re-triggers EWS?

aj062 committed:
Add support for Skip-EWS GitHub label
https://bugs.webkit.org/show_bug.cgi?id=247900

Reviewed by Jonathan Bedard.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin):
(GitHubMixin._does_pr_has_skip_label): Method to check if pr has skip-ews label.
(ValidateChange.__init__):
(ValidateChange):
(ValidateChange.start):
(ValidateChange.validate_github):

Canonical link: https://commits.webkit.org/256760@main
